### ISSUE TITLE
DatetimeIndex + Series[timedelta64] returns DatetimeIndex

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#18963]
This is listed in the roundup #18824 but the behavior has changed since the older issue.

```
dti = pd.date_range('2017-01-01', periods=2)
ser = pd.Series(pd.TimedeltaIndex(['1 Day', '1Day']))
res1 = dti + ser
res2 = ser + dti
res3 = dti - ser
res4 = -ser + dti

>>> res1
DatetimeIndex(['2017-01-02', '2017-01-03'], dtype='datetime64[ns]', freq=None)
>>> res2
0   2017-01-02
1   2017-01-03
dtype: datetime64[ns]
>>> res3
DatetimeIndex(['2016-12-31', '2017-01-01'], dtype='datetime64[ns]', freq=None)
>>> res4
0   2016-12-31
1   2017-01-01
dtype: datetime64[ns]
```

**Comment by jbrockmendel** [comment]
I think there's an easy fix to this in `indexes.datetimelike` dropping a `if isinstance(other, (ABCSeries, ABCDataFrame)): return NotImplemented` right at the top of `__add__` and `__sub__`.

**Comment by jreback** [comment]
seems a bit inconsistent.

**Comment by jbrockmendel** [comment]
Same deal for TimedeltaIndex * Series[int].  In fact this is specifically [tested](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexes/timedeltas/test_arithmetic.py#L55) (it can't be intentional can it?)

```
        idx = TimedeltaIndex(np.arange(5, dtype='int64'))
        didx = TimedeltaIndex(np.arange(5, dtype='int64') ** 2)
        [...]
        result = idx * Series(np.arange(5, dtype='int64'))
        tm.assert_index_equal(result, didx)
```

**Comment by jbrockmendel** [comment]
The OP issue is closed by #18884.  The `TimedeltaIndex * Series[int]` issue mentioned above is still outstanding.
